### PR TITLE
Revert "Revert "Fix canvas issues preventing turtle from drawing line…

### PR DESCRIPTION
**Pull request by davidsbailey**
Reverts code-dot-org/code-dot-org#4218

This is a roll-forward of the rollback of https://github.com/code-dot-org/code-dot-org/pull/4210, with a simple fix for the failing unit tests. Tested with `~/src/cdo/apps$ grunt rebuild test --fast`

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
FWIW, doing a build before test is generally unnecessary. The one exception to this is if you're making local changes. mochaTest ends up creating it's own bundle, which is what it tests against (rather than what you've built).

**Comment by Bjvanminnen** [comment]
lgmt

**Comment by davidsbailey** [comment]
Turns out I broke this test:
  1) Level tests ../solutions/applab/cspunit3.js Expected solution.:
     Uncaught AssertionError: expected [ Array(200) ] to deeply equal [ Array(196) ](http://localhost:52018/js/bundle:104852)

This test seems a little tricky to update. How did you generate the expected value of the pixel array?

**Comment by Bjvanminnen** [comment]
That bit of code could probably use a better comment. Essentially we draw square. I generate the array through a set of for loops. Essentially I go line by line and mark which pixels should be drawn (so line 74 has pixels 100 to 124 inclusive drawn), etc.

**Comment by davidsbailey** [comment]
What do you mean by "making local changes" above? I have local changes to files in `apps/` almost every time I would want to run `grunt test`.

**Comment by Bjvanminnen** [comment]
I meant locale rather than local. Important difference. Basically changes to i18n strings require a rebuild, other stuff shouldnt.

**Comment by davidsbailey** [comment]
@Bjvanminnen I've fixed cspunit3 test. can you please take another quick look?

File: apps/test/solutions/applab/cspunit3.js [hunk 1]
```diff
@@ -144,13 +144,30 @@ module.exports = {
...
+          //  75 0 1 1 1 1...1 1 1 1 0
+          //  76 0 1 1 0 0   0 0 1 1 0
+          //     0 1 1 0 0   0 0 1 1 0
+          //       ...           ...
+          //     0 1 1 0 0   0 0 1 1 0
+          //  98 0 1 1 0 0   0 0 1 1 0
+          //  99 0 1 1 1 1...1 1 1 1 0
+          // 100 0 1 1 1 1...1 1 1 1 0
+          //     0 0 0 0 0   0 0 0 0 0
+          //
```
Comment (Bjvanminnen) [inline apps/test/solutions/applab/cspunit3.js:164]: nice!

**Comment by Bjvanminnen** [comment]
lgtm

**Pull request merged**